openreferee_server/operations.py

```diff
@@ -87,3 +94,32 @@ def cleanup_event(event):
     session = setup_requests_session(event.token)
     cleanup_event_tags(session, event)
     cleanup_file_types(session, event)
+
+
+def process_editable_files(session, files, endpoints):
+    uploaded = defaultdict(list)
+    for _file in files:
+        if _file['filename'].split(".")[-1] == 'pdf':
+            upload = process_pdf(_file['external_download_url'], session, endpoints['file_upload'])
+            uploaded[_file['file_type']].append(upload['uuid'])
+    session.post(endpoints['revisions']['replace'], json={
+        'files': uploaded,
+        'state': 'ready_for_review'
+    })
+
+
+def process_pdf(url, session, upload_endpoint):
+    pdf_writer = PdfFileWriter()
+    rf = urlopen(Request(url, headers=session.headers))
+    pdf_reader = PdfFileReader(io.BytesIO(rf.read()))
+    watermark_pdf = PdfFileReader(open(Path(__file__).parent / 'watermark.pdf', 'rb'))
```

Comment:
`open()` should be used with `with`. But Path objects have something much nicer:

```python
with (Path(__file__).parent / 'watermark.pdf').open('rb') as file:
    watermark_pdf = PdfFileReader(file)
```

Reply:
Right, thanks, I modified this yesterday and completely forgot that.